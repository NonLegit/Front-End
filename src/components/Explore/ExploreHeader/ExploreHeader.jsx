/* eslint-disable import/no-cycle */
import { Typography, Box } from '@mui/material';
import { HeaderContainer, StyledLink } from './styles';
import NavButtons from '../NavButtons/NavButtons';
import { useHeaderSubtitleContext } from '../../../contexts/HeaderSubtitleContext';

/**
 * explore page header
 * @component
 * @property {string} headerSubtitle the header title of the explore page
 * @return {React.Component} - explore page header component
 */

function ExploreHeader(props) {
  const {
    headerSubtitle,
  } = useHeaderSubtitleContext();
  const { title } = props;
  return (
    <HeaderContainer>
      <Box sx={{ padding: '16px 0px' }}>
        <Typography
          fontFamily="sans-serif"
          fontSize="22px"
          color="#1A1A1B"
          fontWeight={600}
        >
          {title}
        </Typography>
        <StyledLink>
          {headerSubtitle}
        </StyledLink>
      </Box>
      <NavButtons title={title} />
    </HeaderContainer>
  );
}

export default ExploreHeader;
