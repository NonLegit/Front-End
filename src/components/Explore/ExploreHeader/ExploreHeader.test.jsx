import renderer from 'react-test-renderer';
import ExploreHeader from './ExploreHeader';
import HeaderSubtitleContextProvider from '../../../contexts/HeaderSubtitleContext';

test('test snapshot', async () => {
  const tree = renderer.create(
    <HeaderSubtitleContextProvider>
      <ExploreHeader />
    </HeaderSubtitleContextProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
