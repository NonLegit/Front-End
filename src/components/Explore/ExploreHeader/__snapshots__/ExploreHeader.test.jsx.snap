// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-3k8df1"
>
  <div
    className="MuiBox-root css-r37xpo"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-5a1xzc-MuiTypography-root"
    />
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1tfr3a3-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      0
    </a>
  </div>
  <div
    className="MuiBox-root css-w8dmq8"
  >
    <div
      className="MuiTabs-root css-1wgiwz8-MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": -0,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-orientation={null}
          className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls={null}
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            Posts
            <span
              className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Communities
          </button>
        </div>
      </div>
    </div>
    <div
      aria-labelledby={null}
      className="MuiTabPanel-root css-13xfq8m-MuiTabPanel-root"
      hidden={false}
      id={null}
      role="tabpanel"
    >
      Posts
    </div>
    <div
      aria-labelledby={null}
      className="MuiTabPanel-root css-y0o6vk-MuiTabPanel-root"
      hidden={true}
      id={null}
      role="tabpanel"
    />
  </div>
</div>
`;
